Tests/DataXScenarios/DataXScenarios/InteractiveQueryAndSchemaGenScenarios.cs

```diff
@@ -0,0 +1,125 @@
+﻿// *********************************************************************
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License
+// *********************************************************************
+using ScenarioTester;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DataX.ServerScenarios
+{
+    /// <summary>
+    /// Partial class defined such that the steps can be defined for the job 
+    /// </summary>
+    public partial class DataXHost
+    {
+        [Step("inferSchema")]
+        public static StepResult InferSchema(ScenarioContext context)
+        {
+            var baseAddress = $"{context[Context.ServiceUrl] as string}/api/DataX.Flow/Flow.SchemaInferenceService/inputdata/inferschema";
+            context[Context.InferSchemaInputJson] = $"{{\"name\": \"{context[Context.FlowName] as string}\", \"userName\": \"{context[Context.FlowName] as string}\", \"eventhubConnectionString\": \"{context[Context.EventhubConnectionString] as string}\", \"eventHubNames\": \"{context[Context.EventHubName] as string}\", \"inputType\": \"iothub\", \"seconds\": \"{context[Context.Seconds] as string}\", \"databricksToken\": \"{context[Context.DataBricksToken] as string}\"}}";
```

Comment:
Can you create a helper function to create this request body?

Reply:
Not refactoring this further. Each step is calling into and using ScenarioTester. Perhaps in a future checkin, a wrapper to ScenarioTester can be created. Leaving this code as is such that the framework for each step is readable and uses the ScenarioTester framework.